src/control/draw.js

```diff
@@ -49,6 +50,30 @@ class DrawControl extends Control {
       style: options.style,
       stopClick: true,
     });
+
+    this.drawInteraction.on('drawstart', () => {
+      this.onDrawStartListener = this.map.on('pointermove', () => {
+        // Sets drawFeature while drawing
+        const drawFeature = this.drawInteraction
+          .getOverlay()
+          .get('source')
+          .getFeatures()
+          .find((feat) => {
+            return (
+              !Array.isArray(feat.getGeometry()) &&
+              feat.getGeometry().getFlatCoordinates().length > 2
+            );
+          });
+        if (drawFeature) {
+          this.editor.setDrawFeature(drawFeature);
```

Comment:
Shouldn't we do that is drawstart event  ? why waiting the pointermove ?

Reply:
Because the drawstart event only provides the feature at its state when you start drawing, it doesn't update the geometry dynamically. With pointermove we get the current geometry of the feature that is being drawn